sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java

```diff
@@ -210,6 +213,20 @@ public String property() {
     public String value() {
       return value;
     }
+
+    @Override
+    public boolean equals(Object o) {
```

Comment:
these `equal/hashCode` are useful in general and we need them in tests. These are generated by IDE, we can also implement these classes in Scala case class to get `equal/hashCode` for free.